src/config_parse.h

```diff
@@ -22,7 +22,7 @@ typedef struct config_file {
 } git_config_file;
 
 typedef struct {
-	git_config_file *file;
+	const char *configname;
```

Comment:
`file_path` ? I think `configname` dilutes the aspect that it's just a filesystem "token" or something…

Reply:
I explicitly wanted to avoid this. Backends may not use files at all, most importantly the memory backend doesn't care for filenames. I thus wanted to have this be generic

Reply:
Our "outer layer" doesn't really know about anything other than files, and will generally needs a "label" that will usually looks like a file path. It's just about labelling that data, be it file-backed or in-memory. `file_identifier`, `config_label`, or something else ?

Really, it's because `configname` sounds like a "configuration preset name", not like "the user-visible name of the configuring thing".

Feel free to keep it as is though, just wanted to see if some bikeshedding would make it read nicer  😉.

Reply:
Fair enough. Would you be happy with `config_identifier`?

Reply:
`config_identifier` it is then. Thanks!

Reply:
Thinking about it some more. Maybe we _should_ just leave this at `file` or `file_path`. At least it seems like an obvious route to go at a later point when implementing in-memory includes.